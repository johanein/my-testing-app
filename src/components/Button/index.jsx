import React, { memo } from "react";
import style from "./style.module.css";

const index = memo(({ label = "" }) => {
  return (
    <div data-testid="button" className={style.button_style}>
      Hi {label}
    </div>
  );
});

export default index;
