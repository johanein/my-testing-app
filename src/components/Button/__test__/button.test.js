import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

import Button from "../index";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Button />, div);
});

it("renders button correctly", () => {
  const { getByTestId } = render(<Button label="click me please" />);
  expect(getByTestId("button")).toHaveTextContent("click me please");
});

it("matches snapshot", () => {
  const tree = renderer.create(<Button label="save" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("matches snapshot", () => {
  const tree = renderer.create(<Button label=" world" />).toJSON();
  expect(tree).toMatchSnapshot();
});
