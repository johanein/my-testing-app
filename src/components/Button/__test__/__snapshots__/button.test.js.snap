// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="button_style"
  data-testid="button"
>
  Hi 
  save
</div>
`;

exports[`matches snapshot 2`] = `
<div
  className="button_style"
  data-testid="button"
>
  Hi 
   world
</div>
`;
